microsoft/community-organization-operations-suite
# Sorting table

**Why** 
- fix #298

**What**
- Created a simple `utils/sortByAlphaNumeric`.
- Created some CSS arrows to display sorting order (ASC or DESC.)
- Added an `onClick` event on the headers of the `ui/PaginatedList` component.
- Fluffed the `ContactList/columns` list with options to make a column sortable.
- Implemented how the click on a column header applies the sorting info.

**Issues**
- First PR to test the water on how the process will work.
- First time working with _React_, _CSS in JS_, and _everything is a component_ methodology.
- Right now, this only works on the **Clients** page as I wanted feedback on code quality and goals.
- Working on using the Translation utility outside of the React Component Hook lifecycle.
- Still figuring out how column sorting could be leveraged on the **Reporting** page.

**Testing**
 - Go to the **Clients** page and try sorting some columns.
 - Make sure that only the following columns are sortable (`Name`, `Date of Birth`, and `Gender`.)

**Screenshot**

https://user-images.githubusercontent.com/636801/152589175-73f85aa2-1fdd-415c-8a66-96b7131248fe.mov
